File: get_sha.py
import subprocess
import os

keytool_path = r'C:\Program Files\Android\Android Studio\jbr\bin\keytool.exe'
keystore_path = r'C:\Users\Manuel\.android\debug.keystore'

cmd = [
    keytool_path,
    '-list', '-v',
    '-keystore', keystore_path,
    '-alias', 'androiddebugkey',
    '-storepass', 'android',
    '-keypass', 'android'
]

print(f"Running: {' '.join(cmd)}")

try:
    result = subprocess.run(cmd, capture_output=True, text=True, check=True)
    print("--- SUCCESS ---")
    print(result.stdout)
except subprocess.CalledProcessError as e:
    print("--- ERROR ---")
    print(e.stdout)
    print(e.stderr)
except Exception as e:
    print(f"--- FAILURE ---: {e}")

File: get_sha.dart
import 'dart:io';

void main() async {
  final keytoolPat = r'C:\Program Files\Android\Android Studio\jbr\bin\keytool.exe';
  final keystorePath = r'C:\Users\Manuel\.android\debug.keystore';
  
  print('Iniciando extracción con: $keytoolPat');
  print('Ruta del KeyStore: $keystorePath');

  try {
    final result = await Process.run(keytoolPat, [
      '-list',
      '-v',
      '-keystore', keystorePath,
      '-alias', 'androiddebugkey',
      '-storepass', 'android',
      '-keypass', 'android'
    ]);

    if (result.exitCode == 0) {
      print('--- RESULTADO ---');
      print(result.stdout);
    } else {
      print('--- ERROR ---');
      print(result.stderr);
      print('Exit code: ${result.exitCode}');
    }
  } catch (e) {
    print('Error ejecutando el proceso: $e');
  }
}
